src/coreComponents/linearAlgebra/DofManager.hpp

```diff
@@ -71,6 +71,25 @@ class DofManager
     string meshLevelName;
     /// list of the region names
     std::vector< string > regionNames;
+
+    bool add( Regions const & input )
+    {
+      bool added = false;
+      if( meshBodyName  == input.meshBodyName && meshLevelName == input.meshLevelName )
+      {
+        for( auto const & inputRegionName : input.regionNames )
+        {
+          if( std::find( regionNames.begin(), regionNames.end(), inputRegionName ) == regionNames.end() )
+          {
+            regionNames.push_back( inputRegionName );
```

Comment:
If you want to make the insertion into `regionNames` specific, maybe it would make sense to declare `regionNames` `private` to impose the usage of `add`?
Plus maybe renaming into `addRegionName` would be a tad clearer? (plus doxygen of course)

Reply:
hmm would have to check how else it's used though. I mean I can do it but it may need a lot more changes.